src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -27,6 +86,85 @@ private static void VerifyNoOverride(CSharpCompilation comp, string methodName)
+        {
+            foreach (var tree in comp.SyntaxTrees)
+            {
+                var model = comp.GetSemanticModel(tree);
+                foreach (var assignment in tree.GetRoot().DescendantNodes().OfType<AssignmentExpressionSyntax>())
+                {
+                    var right = assignment.Right;
+                    var typeInfo = model.GetTypeInfo(right);
+                    Assert.True(typeInfo.Type.GetSymbol<TypeSymbol>().Equals(typeInfo.ConvertedType.GetSymbol<TypeSymbol>(), TypeCompareKind.AllIgnoreOptions));
+                }
+            }
+        }
+
+        private CSharpCompilation CreateCompilationWithCovariantReturns(string source, MetadataReference[] references = null, TargetFramework targetFramework = TargetFramework.Standard)
+        {
+            return CreateCompilation(source, parseOptions: TestOptions.WithCovariantReturns, references: references, targetFramework: targetFramework);
+        }
+
+        private CSharpCompilation CreateCompilationWithoutCovariantReturns(string source, MetadataReference[] references = null, TargetFramework targetFramework = TargetFramework.Standard)
+        {
+            return CreateCompilation(source, parseOptions: TestOptions.WithoutCovariantReturns, references: references, targetFramework: targetFramework);
+        }
+
+        private static CSharpCompilation CompilationReferenceView(CSharpCompilation comp, params MetadataReference[] references)
+        {
+            references = references.Append(comp.ToMetadataReference());
+            return CreateCompilation("", references: references);
+        }
+
+        private static CSharpCompilation MetadataView(CSharpCompilation comp, params MetadataReference[] references)
+        {
+            references = references.Append(comp.EmitToImageReference());
+            return CreateCompilation("", references: references);
+        }
+
+        [Fact]
+        public void CovariantReturns_00()
+        {
+            var source = @"
+public class Base
+{
+    public virtual string M() => null;
+}
+public class Derived : Base
+{
+    public override string M() => null;
+}
+public class Program
+{
+    void M(Base b, Derived d)
+    {
+        string s1 = b.M();
+        string s2 = d.M();
+    }
+}
+";
+            var comp = CreateCompilationWithoutCovariantReturns(source).VerifyDiagnostics(
+                );
+            verify(comp);
+            verify(CompilationReferenceView(comp));
+            verify(MetadataView(comp));
+
+            comp = CreateCompilationWithCovariantReturns(source).VerifyDiagnostics(
+                );
+            verify(comp);
+            verify(CompilationReferenceView(comp));
+            verify(MetadataView(comp));
+
+            static void verify(CSharpCompilation comp)
+            {
+                VerifyOverride(comp, "Derived.M", "System.String Derived.M()", "System.String Base.M()");
+                VerifyAssignments(comp);
```

Comment:
>VerifyAssignments [](start = 16, length = 17)

It doesn't look like there are assignment expressions in the tree. Is this call going to verify anything?

Reply:
In the following PR the helper verifies local declaration initializers.

---
In reply to: [422131517](https://github.com/dotnet/roslyn/pull/43797#discussion_r422131517) [](ancestors = 422131517)